app/saml.server.ts

```diff
@@ -0,0 +1,54 @@
+// saml server based on following PR
+// https://github.com/remix-run/examples/pull/130/files/ec66b3060fac83eec2389eb0c96aad6d8ea4aed1#diff-02d2b71e481b2495b8a72af14f09fc28238298c7f1d19a540e37c9228985b0da
+import * as samlify from 'samlify';
+import * as validator from '@authenio/samlify-node-xmllint';
+import { readFileSync } from 'fs';
+import secrets from './lib/secrets.server';
+
+samlify.setSchemaValidator(validator);
+
+// Here we configure the service provider: https://samlify.js.org/#/sp-configuration
+
+const sp = samlify.ServiceProvider({
+  entityID: process.env.SAML_ENTITY_ID,
+  nameIDFormat: ['urn:oasis:names:tc:SAML:2.0:nameid-format:persistent'],
+  wantAssertionsSigned: true,
+  assertionConsumerService: [
+    {
+      Binding: 'urn:oasis:names:tc:SAML:2.0:bindings:HTTP-POST',
+      Location: process.env.HOSTNAME + '/login/callback',
+    },
+  ],
+  singleLogoutService: [
+    {
+      Binding: 'urn:oasis:names:tc:SAML:2.0:bindings:HTTP-POST',
+      Location: process.env.HOSTNAME + '/logout/callback',
+    },
+  ],
+});
+
+// Take the metadata stood up by the IDP and use it as the metadata for our IDP object
+const idp = samlify.IdentityProvider({
+  metadata: readFileSync(secrets.SAML_IDP_METADATA_FILE),
```

Comment:
Three things:

1. You don't need to `readFileSync()` on the `secrets`, they are already re-hydrated from the filesystem
2. Rename it to `SAML_IDP_METADATA` (drop the `_FILE` since they are all files)
3. You should add some code at the top of this to check that this secret exists, and throw if not.  We can't run the app without it.

```ts
const { SAML_IDP_METADATA } = secrets;
if(!SAML_IDP_METADATA) {
  throw new Error('Missing SAML_IDP_METADATA secret');
}
```